java/src/test/java/ai/rapids/cudf/DecimalColumnVectorTest.java

```diff
@@ -0,0 +1,187 @@
+/*
+ *  Copyright (c) 2019-2020, NVIDIA CORPORATION.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package ai.rapids.cudf;
+
+import ai.rapids.cudf.HostColumnVector.Builder;
+import org.junit.jupiter.api.Test;
+
+import java.math.BigDecimal;
+import java.util.Random;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class DecimalColumnVectorTest extends CudfTestBase {
+  private final Random rdSeed = new Random();
+
+  private final BigDecimal[] decimal32Zoo = new BigDecimal[]{
+    BigDecimal.valueOf(rdSeed.nextInt(1000000000), 4),
+    BigDecimal.valueOf(rdSeed.nextInt(1000000000), 4),
+    BigDecimal.valueOf(rdSeed.nextInt(1000000000), 4),
+  };
+
+  private final BigDecimal[] decimal64Zoo = new BigDecimal[]{
+    BigDecimal.valueOf(rdSeed.nextLong(), 10),
+    BigDecimal.valueOf(rdSeed.nextLong(), 10),
+    null,
+    BigDecimal.valueOf(rdSeed.nextLong(), 10),
+  };
+
+  @Test
+  public void testCreateColumnVectorBuilder() {
+    try (ColumnVector decColumnVector = ColumnVector.build(DType.create(DType.DTypeEnum.DECIMAL32, -5), 3,
+        (b) -> b.append(BigDecimal.valueOf(123456789, 5)))) {
+      assertFalse(decColumnVector.hasNulls());
+    }
+    try (ColumnVector decColumnVector = ColumnVector.build(DType.create(DType.DTypeEnum.DECIMAL64, -10), 3,
+        (b) -> b.append(BigDecimal.valueOf(1023040506070809L, 10)))) {
+      assertFalse(decColumnVector.hasNulls());
+    }
+  }
+
+  @Test
```

Comment:
There needs to be a negative test verifying the code throws the expected exception if the user tries to add a decimal that cannot be represented in cudf.

Reply:
A case of precision overflow has been added at [here](https://github.com/rapidsai/cudf/pull/6609/files#diff-e7430fdc52346d6e5fafaf008865f29e50d7e3ee163e09b93cc1fc1a9c8d1ab6R118). I am not sure whether it is adequate.